src/puppetlabs/pcp/client.clj

```diff
@@ -325,6 +325,14 @@
     (.start client)
     client))
 
+(defn ^:private append-client-type
+  "Append client type to a ws connection Uri, accounting for possible trailing
+   slash."
+  [url type]
+  (if (= (last url) \/)
```

Comment:
I'd probably use java's `String#endsWith` to avoid creation & iteration through the collection of characters.